tests/rest/client/v2_alpha/test_register.py

```diff
@@ -130,3 +131,24 @@ def test_POST_disabled_guest_registration(self):
 
         self.assertEquals(channel.result["code"], b"403", channel.result)
         self.assertEquals(channel.json_body["error"], "Guest access is disabled")
+
+    def test_POST_ratelimiting(self):
+        self.hs.config.rc_registration_request_burst_count = 5
+
+        for i in range(0, 6):
+            url = self.url + b"?kind=guest"
```

Comment:
I would be tempted to have a second test case where we didn't try and register as guests